Choose I²C Address

Granjow opened the pull request:
The LD pressure sensors have a configurable I²C address. This MR allows to use the sensors even if they use a different address, and thus would also allow to interface with multiple sensors on the same I²C bus.

Additionally, it adds a PlatformIO file which is helpful for development as the project can easily be opened with CLion, VS Code, etc.

Granjow committed:
Re-arrange for PlatformIO support

Granjow committed:
Fix typo

Granjow committed:
Allow custom I²C address, rename calibration date variables

Granjow commented:
@patrickelectric Any chance this MR is going to be merged? If yes, I will fix the conflict.

patrickelectric commented on platformio.ini:
```diff
@@ -0,0 +1,15 @@
+[env:promini]
+platform = atmelavr
+board = pro8MHzatmega328
+framework = arduino
```
This file is also not necessary.

patrickelectric commented on .gitignore:
```diff
@@ -0,0 +1,5 @@
+.idea
+CMake*
+CMakeListsPrivate.txt
+cmake-build-*/ 
```
This file is unnecessary, we don't use cmake in this project. Also the others formats feel unimportant. Usually .gitignore files are used to put generated files by the project or build tools. And I don't see the case here for this project.

patrickelectric commented on src/KellerLD.cpp:
```diff
@@ -11,8 +11,13 @@
-KellerLD::KellerLD() {
-	fluidDensity = 1029;
+
+KellerLD::KellerLD()
+		: i2cAddress(LD_DEFAULT_ADDR) {
+}
+
+KellerLD::KellerLD(int i2cAddress)
+		: i2cAddress(i2cAddress) {
 }
```
```suggestion
KellerLD::KellerLD(int i2cAddress = LD_DEFAULT_ADDR)
		: i2cAddress(i2cAddress) {
}
```

patrickelectric reviewed (changes requested):
Hi @Granjow, sorry for taking this long to review, it got somehow lost in my list.
Be free to ping me in any PR or issue if I do not reply in the desirable time.

Granjow committed:
Simplify constructor: Use default arg

Co-authored-by: Patrick José Pereira <patrickelectric@gmail.com>

Granjow committed:
Remove CMake entries from .gitignore

Granjow commented:
Thanks @patrickelectric  for the review!

I added some files (platformio.ini and main.cpp) because it greatly simplifies development. I can [open the platformio.ini file in CLion](https://www.jetbrains.com/help/clion/platformio.html#create-prj) and I have full IDE support; and because of the main.cpp I can build the project and see if there are build errors. I assume it is the same for VSCode. It would also allow to use a CI pipeline for automated testing (at least checking for build failures). I'm not familiar with GitHub, but it should be a matter of minutes to set up CI with PlatformIO.

This is why I'd keep the files in the repo, but if you want to have them deleted, I can do that.

CMake was an artifact from an earlier version of the PlatformIO plugin for CLion. I removed it.

Granjow committed:
Update changelog for 1.1.2

patrickelectric commented:
Hi @Granjow, sorry for losing track of this PR, can you rebase over master ?

Granjow commented:
Hi @patrickelectric , done!

vshie commented:
@patrickelectric forget to merge?
